plugins/out_s3/s3.c

```diff
@@ -1070,13 +1102,19 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
             if (chunk) {
                 s3_store_file_unlock(chunk);
             }
+            if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
+                flb_free(payload_buf);
+            }
```

Comment:
@zhonghui12 technically it was fine to call free on NULL before when the code didn't have this if condition... `free(NULL)` doesn't do anything...